### PR TITLE
Fix start.py and Grafana dashboard

### PR DESCRIPTION
**Pull request by amandahla**
### Overview

This PR fixes two issues:
- Remove chown from original start.py script, preventing the refresh of taking too long because /data is a big directory and chown takes a long time to run.
- Fix CPU usage Grafana dashboard. Maybe is related to some prometheus_client upgrade in Synapse but after the refresh the graph was different.

### Rationale

Fix issues.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

No need of changing charmhub doc

### CODE REVIEW

File: synapse_rock/start_custom.py
Comment (cbartz) [inline synapse_rock/start_custom.py:1]: https://github.com/element-hq/synapse/blob/develop/docker/start.py could be updated in the future, what are the plans to keep it up to date?

File: synapse_rock/rockcraft.yaml [hunk 1]
```diff
@@ -129,4 +126,18 @@ parts:
...
-            login_ubuntu.py: /usr/local/attributemaps/login_ubuntu.py
\ No newline at end of file
+            login_ubuntu.py: /usr/local/attributemaps/login_ubuntu.py
+    synapse-start:
+        plugin: dump
+        after:
+          - synapse
+        source: .
+        # start.py is extracted from
+        # https://github.com/matrix-org/synapse/blob/develop/docker/start.py
```
Comment (cbartz) [inline synapse_rock/rockcraft.yaml:136]: You may want to point to https://github.com/element-hq/synapse/blob/develop/docker/start.py as the other repo is a public archive.

File: src/grafana_dashboards/synapse.json
Comment (cbartz) [inline src/grafana_dashboards/synapse.json:1]: Does the dashboard look different after the change? It is difficult to tell from the diff.

File: synapse_rock/rockcraft.yaml [hunk 2]
```diff
@@ -129,4 +126,18 @@ parts:
...
+        # Our version does not use chown command that, depending of the /data
+        # size, could take a long time to run.
+        organize:
+            start_custom.py: start.py
```
Comment (javierdelapuente) [inline synapse_rock/rockcraft.yaml:140]: Would it be better to "patch" the original one, so we can see when it is changed upstream?

**Comment by mthaddon** [comment]
What's the implication of not doing the chown? Could that cause problems for us later?

File: src/grafana_dashboards/synapse.json
Comment (amandahla) [reply src/grafana_dashboards/synapse.json:1]: The main difference is migrating from graph (deprecated) to timeseries.

File: synapse_rock/rockcraft.yaml (hunk 2 above)
Comment (amandahla) [reply synapse_rock/rockcraft.yaml:140]: Do you mean patching in the rockcraft itself? Using sed commands or something similar?

**Comment by amandahla** [comment]
> What's the implication of not doing the chown? Could that cause problems for us later?

I thought that charm storage was mounted with 777 permission but that's not the case. So I'll change the command from 'chown' to 'find . -maxdepth 1 -exec chown {ownership} "{}" \;'
When Synapse starts for the first time creates just this:
```
homeserver.db  homeserver.db-shm  homeserver.db-wal  homeserver.yaml  log.config  lost+found  media  test.macaroon.key  test.registration.key  test.signing.key  worker.yaml
```

So changing the command from chown to find will impact only in the case the user was changed and media directory content still has the old permission.

File: synapse_rock/rockcraft.yaml (hunk 1 above)
Comment (amandahla) [reply synapse_rock/rockcraft.yaml]: Removed ;)

File: src/grafana_dashboards/synapse.json
Comment (amandahla) [reply src/grafana_dashboards/synapse.json:1]: I'll mark as resolved, feel free to re-open it.

File: synapse_rock/rockcraft.yaml (hunk 2 above)
Comment (amandahla) [reply synapse_rock/rockcraft.yaml]: But then I would need to git clone synapse repository right? Right now the start.py is obtained from what pip installs.

File: synapse_rock/start_custom.py
Comment (amandahla) [reply synapse_rock/start_custom.py:1]: Javier suggested using a patch so now if for some reason it fails we will know during the rock build.

**Review by merkata — APPROVED** [review]
Nice!

**Review by javierdelapuente — APPROVED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Review by merkata — APPROVED** [review]

**Review by merkata — DISMISSED** [review]

File: synapse_rock/patches/start.patch [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
++#!/usr/bin/python3
+ 
+ import codecs
+ import glob
+@@ -136,7 +136,7 @@
+ 
+     if ownership is not None:
+         log(f"Setting ownership on /data to {ownership}")
+-        subprocess.run(["chown", "-R", ownership, "/data"], check=True)
++        subprocess.run(["find", "/data", "-maxdepth", "1", "-exec", "chown",ownership, "{}", ";"], check=True)
```
Comment (codethulu) [inline synapse_rock/patches/start.patch:14]: should we have space after "chown"? small nitpick i know...

**Review by codethulu — APPROVED** [review]

**Comment by amandahla** [comment]
"tests/integration/test_s3.py:104:65: E1135: Value 'magic.from_buffer(objbuf)' doesn't support membership test (unsupported-membership-test) 

[22](https://github.com/canonical/synapse-operator/actions/runs/9081358604/job/24955027561?pr=337#step:6:23)"
???

**Review by codethulu — APPROVED** [review]

**Pull request merged**